.github/workflows/release.yml

```diff
@@ -5,82 +5,175 @@ permissions:
 
 on:
   workflow_dispatch:
+    inputs:
+      force:
+        required: false
+        description: 'If non-empty, extra force will be used push this release'
+        default: ''
+      commit:
+        required: false 
+        description: 'Fluvio git commit override (latest `master` by default)'
+        default: ''
+env:
+  USE_COMMIT: ${{ github.event.inputs.commit }}
+  FORCE_RELEASE: ${{ github.event.inputs.force }}
 
 jobs:
+  setup_job:
+    name: Setup workflow
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version_step.outputs.VERSION }}
+      target_sha: ${{ steps.version_step.outputs.GIT_SHA }}
+      latest_tag: ${{ steps.docker_step.outputs.LATEST_TAG }}
+      release_tag: ${{ steps.docker_step.outputs.RELEASE_TAG }}
+    steps:
+      - name: Set target sha and Fluvio version
+        id: version_step
+        run: |
+          if [[ -z "${USE_COMMIT}" ]]; then
+            echo "VERSION=$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.sha }}/VERSION)" >> $GITHUB_ENV
+            echo "::set-output name=VERSION::$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.sha }}/VERSION)"
+            echo "GIT_SHA=${{ github.sha }}" >> $GITHUB_ENV
+            echo "::set-output name=GIT_SHA::${{ github.sha }}"
+          else
+            echo "VERSION=$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.event.inputs.commit }}/VERSION)" >> $GITHUB_ENV
+            echo "::set-output name=VERSION::$(curl -fsS https://raw.githubusercontent.com/infinyon/fluvio/${{ github.event.inputs.commit }}/VERSION)"
+            echo "GIT_SHA=${{ github.event.inputs.commit }}" >> $GITHUB_ENV
+            echo "::set-output name=GIT_SHA::${{ github.event.inputs.commit }}"
+          fi
+      - name: Set Docker tag related env vars
+        id: docker_step
+        run: |
+            echo "::set-output name=LATEST_TAG::infinyon/fluvio:${{ env.VERSION }}-${{ env.GIT_SHA }}"
+            echo "::set-output name=RELEASE_TAG::infinyon/fluvio:${{ env.VERSION }}"
+
+  # Check for Github Release
   release_github:
-    name: Create GitHub Release
+    name: Release Fluvio to GitHub Release
+    needs: [setup_job]
```

Comment:
Are you sure that environment settings are passed from one job to another? I'm pretty sure it works that way for _steps_, but I have not yet tried this with a job.needs

Reply:
Yeah, I'm sure after some trial and error. I linked some jobs from my dev repo to test it out.

I believe I found that exported vars don't pass from job to job. However, you can use `set-output` to pass data between jobs.